### DISCUSSION TITLE
Default Filtering on Models

### DISCUSSION THREAD

**Discussion by eraffel-MDSol** [jeremyevans/sequel#1985] [General]
Is there a way to have filtering executed by default for a model? For example, let's say the table has a boolean property, `soft_deleted`, and I want the model to only return records where `soft_deleted` is false when querying the model

**Comment by jeremyevans** [comment]
```ruby
class Foo < Sequel::Model(DB[:foos].exclude(:soft_deleted))
end
```

**Reply by eraffel-MDSol** [reply]
Thanks for replying so quickly. I should have been more specific to my needs. Is there a way to make this a condition that is executed at query time? I want the condition to be based on a value in Thread.current that will change per-request

**Reply by jeremyevans** [reply]
You can use `Sequel.delay` for that (http://sequel.jeremyevans.net/rdoc/classes/Sequel/SQL/Builders.html#method-i-delay).

**Reply by AlexWayfer** [reply]
Even a block, passed there to `.where` will be executed immediately, at the class defenition step, as I understand.

So, I'd suggest to define a simple dataset method, which will add this filtration at the invokation moment. A little more code, but the architecture condition is strange for me.

**Reply by eraffel-MDSol** [reply]
I think that worked with the delay. Thanks so much!

The architecture is that I'm trying to set a global filtering scope that will apply to multiple classes based on a flag, and I'm trying to not have it everywhere in the application code.
